### PR TITLE
Adding Safe Mode

### PR DESCRIPTION
**Pull request by IronCore864**
Changes:

- Add a `--safe-mode` boolean option
  - If enabled, script exits on aws exception/error. E.g., wrong key/secret would exit directly without generating new config file.
  - Also, checks the instance list in each security grouop. If at least one of the security groups is empty, i.e., no instance in one or more security groups, new haproxy config will be generated, but not relaoded.
  - Default `False` so backward compatible.
- Add timestamp into logs, and redirect logs to STDOUT.
In this case if you pipe the output to other command like `grep`, it works.
- PEP8 reformat.
- Add .idea folder into .gitignore, which is used by pycharm

### CODE REVIEW

File: haproxy_autoscale/haproxy_autoscale.py [hunk 1]
```diff
@@ -7,6 +7,7 @@
 
 __version__ = '0.4.1'
```
Comment (markcaudill) [inline haproxy_autoscale/haproxy_autoscale.py]: It looks like I missed this in the last PR, but could you update `__version__` and resubmit the PR?

**Comment by IronCore864** [comment]
Version updated.
Also update the changelog in the readme to make it the same as the version.

**Comment by IronCore864** [comment]
Previous suggestions updated.

In the latest commit I also added a new feather which is the `--delay` option.

If specified, for example `--delay 120` (seconds), instances that were just created within two minutes won't be added into haproxy conf.

This is useful when you want to give it some time to make sure the service on the instance is up and running before adding it into haproxy. Backward compatibility is guaranteed.

**Comment by markcaudill** [comment]
Thanks again for all these changes! I think it really adds to the project.

**Pull request merged**